src/core/kdtree.hpp

```diff
@@ -0,0 +1,323 @@
+/*
+* This file is part of OpenTTD.
+* OpenTTD is free software; you can redistribute it and/or modify it under the terms of the GNU General Public License as published by the Free Software Foundation, version 2.
+* OpenTTD is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
+* See the GNU General Public License for more details. You should have received a copy of the GNU General Public License along with OpenTTD. If not, see <http://www.gnu.org/licenses/>.
+*/
+
+/** @file kdtree.hpp K-d tree template specialised for 2-dimensional Manhattan geometry */
```

Comment:
Possibly `kdtree` is the wrong name in this case? Might give people the wrong idea of what it is. But I have no real suggestions :D `kdtree_2d_manhattan` :P

Reply:
The general description of this data structure and associated algorithms is called k-dimensional tree, so even if the implementation is specialised for a fixed number of dimensions I think it's still appropriate to use that name.